### PR TITLE
Improve slack-bot-ci lambda

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Move the code to `tests/ci`; add deployment script and dependencies.

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit 4d0d16174867a62ce9272eda5f402613382c42de with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54258/4d0d16174867a62ce9272eda5f402613382c42de/ci_running.html)
The overall status of the commit is 🟢 success

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟢 success</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: tests/ci/slack_bot_ci_lambda/app.py [hunk 1]
```diff
@@ -1,28 +1,26 @@
...
 
-# A trivial stateless slack bot that notifies about new broken tests in ClickHouse CI.
-# It checks what happened to our CI during the last check_period hours (1 hour) and notifies us in slack if necessary.
-# This script should be executed once each check_period hours (1 hour).
-# It will post duplicate messages if you run it more often; it will lose some messages if you run it less often.
-#
-# You can run it locally with no arguments, it will work in a dry-run mode. Or you can set your own SLACK_URL_DEFAULT.
-# Feel free to add more checks, more details to messages, or better heuristics.
-# NOTE There's no deployment automation for now,
-# an AWS Lambda (slack-ci-bot-test lambda in CI-CD) has to be updated manually after changing this script.
```
Comment (tavplubix) [inline tests/ci/slack_bot_ci_lambda/app.py:11]: Please add the name back to the comment. So you will not be the only one who knows how and where we run it

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
Well, I did not expect that robot would merge it

File: tests/ci/slack_bot_ci_lambda/app.py [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
 import json
 import base64
 import random
 
-if os.environ.get("AWS_LAMBDA_ENV", "0") == "1":
-    # For AWS labmda (python 3.7)
-    from botocore.vendored import requests
-else:
-    # For running locally
-    import requests
```
[the remark this answers is not in the document]
Comment (Felixoid) [reply tests/ci/slack_bot_ci_lambda/app.py:25]: Yes, all lambdas are deployed by https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/team_keys_lambda/build_and_deploy_archive.sh

It creates a zip archive with all the dependencies described in `requirements.txt`

And `requests` is a dependency in `lambda_shared_package/pyproject.toml`. There's a simplified [ClickHouseHelper](https://github.com/ClickHouse/ClickHouse/blob/3a07b80f3415c01aa20ddb5cf61abc4a03c8d4bb/tests/ci/lambda_shared_package/lambda_shared/__init__.py#L44) btw, so maybe you'd like to use it later

**Comment by Felixoid** [comment]
Yeah... it was unexpected

**Pull request merged**